### PR TITLE
[IOTDB-2866] Support get or create SchemaPartition with PatternTree in config node

### PR DESCRIPTION
**Pull request by CRZbulabula**
changes:

- Support PatternTree in ConfigNode
- Normalize thrift interfaces. See [ConfigNode for DataNode RPC 接口定义](https://y8dp9fjm8f.feishu.cn/docs/doccnCp13O7QDU7O6GMvagUwEDc#), [common.thrift 定义](https://y8dp9fjm8f.feishu.cn/docs/doccnwYEhSIw2a1Qo1F2KXsfLIb#zSavsx) for more details.

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataPartitionDataSet.java [hunk 1]
```diff
@@ -97,14 +98,12 @@ public void convertToRpcDataPartitionResp(TDataPartitionResp resp) {
...
-                                      regionReplicaSet.getConsensusGroupId().getId());
-
-                                  // Set TRegionReplicaSet's GroupType
-                                  tRegionReplicaSet.setGroupType("DataRegion");
-
+                                  ByteBuffer buffer =
+                                      ByteBuffer.allocate(Byte.BYTES + Integer.BYTES);
+                                  regionReplicaSet.getConsensusGroupId().serializeImpl(buffer);
+                                  buffer.flip();
+                                  tRegionReplicaSet.setRegionId(buffer);
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataPartitionDataSet.java:106]: why?

**Review by qiaojialin — APPROVED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java [hunk 2]
```diff
@@ -123,26 +132,94 @@ public TSStatus setStorageGroup(PhysicalPlan physicalPlan) {
   }
 
   @Override
-  public DataSet getSchemaPartition(PhysicalPlan physicalPlan) {
+  public DataSet getSchemaPartition(PathPatternTree patternTree) {
+    TSStatus status = confirmLeader();
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java:136]: getSchemaPartition should connect all node ,include follwer node.   follwer node return result.

File: confignode/src/main/java/org/apache/iotdb/confignode/physical/PhysicalPlan.java [hunk 3]
```diff
@@ -108,7 +108,7 @@ public static PhysicalPlan create(ByteBuffer buffer) throws IOException {
           plan = new GetOrCreateSchemaPartitionPlan(PhysicalPlanType.GetSchemaPartition);
           break;
         case CreateSchemaPartition:
-          plan = new CreateSchemaPartitionPlan(PhysicalPlanType.CreateSchemaPartition);
+          plan = new CreateSchemaPartitionPlan();
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/physical/PhysicalPlan.java:111]: should be GetSchemaPartitionPlan?

Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/physical/PhysicalPlan.java:111]: I distinguish the operations between query and write about PartitionTable in ConfigNode. When DataNode send a getOrCreate request, ConfigNode first check if there exists Partitions that haven't been assigned. And ConfigNode will use create plan to allocate these non-assigned Partitions.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java:136]: For now, we first give all read and write requests to the leader for execution, and then further optimize with the consensus layer.

File: confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataPartitionDataSet.java (hunk 1 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/DataPartitionDataSet.java:106]: We redefine the regionId in TRegionReplicaSet. You can see [commmon.thrift](https://y8dp9fjm8f.feishu.cn/docs/doccnwYEhSIw2a1Qo1F2KXsfLIb#zSavsx) for more details~

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**